### DISCUSSION TITLE
What does `error: 'spi-builder-workspace': Invalid manifest` mean?

### DISCUSSION THREAD

**Discussion by heckj** [SwiftPackageIndex/SwiftPackageIndex-Server#2745] [Q&A] [answered]
Morning - I was digging around in my build logs (example in question: https://swiftpackageindex.com/builds/967AC3B8-B943-4910-B7BD-27AA3EBEF53B) and saw that Swift 5.7 - in particular - (not Swift 5.6 or Swift 5.8) was failing to build for [this package](https://swiftpackageindex.com/heckj/CRDT). 

The swift error is that when it's parsing Package.swift, it can't resolve the symbol for `ProcessInfo`, which I'm using within the package to adjust the settings based on environment variables. But while I was noodling on that, I noticed the error above it:

```
error: 'spi-builder-workspace': Invalid manifest
```

And wasn't sure what was happening. Is there a builder that's gone awry, or a flaw in an .spi file? It's not happening with other versions of builders (Swift 5.8, etc) - but I don't know how they're segmented, and what might be happening with this error.

**Comment by finestructure** [comment, marked as answer]
I can't check right now but shouldn't it also do `import Foundation` and not just check `canImport`?

```swift
  #if canImport(Foundation)
    import Foundation // this here?
    if ProcessInfo.processInfo.environment["CI"] != nil {
        globalSwiftSettings.append(.unsafeFlags(["-Xfrontend", "-strict-concurrency=complete"]))
```

**Reply by finestructure** [reply]
(no idea why that would happen only on 5.7 though, if that's indeed the problem)

**Reply by heckj** [reply]
The #if canImport() nonsense was a vague attempt to see if i was making a crap assumption on Swift 5.7 - it's working on all the other versions, which is really throwing me sideways. But yeah, I should just explicitly import Foundation and see how it rolls.

**Comment by heckj** [comment]
Adding the explicit "import Foundation" sorted the whole thing for Swift 5.7 - lord knows why only that one, and not Swift 5.6 or Swift 5.8 or later. I guess there was a quirky scenario where Foundation wasn't implicitly imported (it typically is on macOS, but not on iOS or Linux)
